Makefile

```
@@ -77,7 +79,9 @@ client: generate
 	hack/build-client.sh "${AUTOREST_IMAGE}" 2020-04-30 2021-09-01-preview 2022-04-01 2022-09-04 2023-04-01 2023-07-01-preview 2023-09-04 2023-11-22 2024-08-12-preview
 
 ci-rp: fix-macos-vendor
-	docker build . -f Dockerfile.ci-rp --ulimit=nofile=4096:4096 --build-arg REGISTRY=$(REGISTRY) --build-arg ARO_VERSION=$(VERSION) --no-cache=$(NO_CACHE)
+	docker build . -f Dockerfile.ci-rp --ulimit=nofile=4096:4096 --build-arg REGISTRY=$(REGISTRY) --build-arg ARO_VERSION=$(VERSION) --target portal-build --no-cache=$(NO_CACHE) -t $(ARO_PORTAL_BUILD_IMAGE)
+	docker build . -f Dockerfile.ci-rp --ulimit=nofile=4096:4096 --build-arg REGISTRY=$(REGISTRY) --build-arg ARO_VERSION=$(VERSION) --target builder --cache-from portal-build -t $(ARO_BUILDER_IMAGE)
```

https://docs.docker.com/reference/cli/docker/image/build/#cache-from
https://docs.podman.io/en/stable/markdown/podman-build.1.html
According to the doc, `--cache-from` should point to an image name.
I'm not sure if it works with the stage name (maybe not?).

And also it wants to have `--no-cache` here, otherwise it can use the cache if the previous build has the same result.

```suggestion
	docker build . -f Dockerfile.ci-rp --ulimit=nofile=4096:4096 --build-arg REGISTRY=$(REGISTRY) --build-arg ARO_VERSION=$(VERSION) --target builder --no-cache=$(NO_CACHE) --cache-from $(ARO_PORTAL_BUILD_IMAGE) -t $(ARO_BUILDER_IMAGE)
```

---

I was wrong about `--no-cache`. Please forget about it.

---

Okay no worries.

---

Updated --cache-from to point to an image name